File: aide/core/output.py
"""输出格式工具，统一使用前缀符号。"""


def ok(message: str) -> None:
    print(f"✓ {message}")


def warn(message: str) -> None:
    print(f"⚠ {message}")


def err(message: str) -> None:
    print(f"✗ {message}")


def info(message: str) -> None:
    print(f"→ {message}")


def step(message: str, current: int | None = None, total: int | None = None) -> None:
    if current is not None and total is not None:
        print(f"[{current}/{total}] {message}")
    else:
        print(f"[·] {message}")

File: src/core/output.rs
use std::io::{self, Write};

pub fn ok(message: &str) {
    println!("\u{2713} {message}");
}

pub fn warn(message: &str) {
    println!("\u{26A0} {message}");
}

pub fn err(message: &str) {
    let _ = writeln!(io::stderr(), "\u{2717} {message}");
}

pub fn info(message: &str) {
    println!("\u{2192} {message}");
}

#[allow(dead_code)]
pub fn step(message: &str, current: Option<u32>, total: Option<u32>) {
    match (current, total) {
        (Some(c), Some(t)) => println!("[{c}/{t}] {message}"),
        _ => println!("[\u{00B7}] {message}"),
    }
}
